### PR TITLE
Simplify `dependency()` reporting

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: tests/testthat/_snaps/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    Code
+      dependencies(errors = "fatal")
+    Output
+      Finding R package dependencies ... Done!
+      WARNING: One or more problems were discovered while enumerating dependencies.
+      
+      renv-test-ecd07ec4e426
+      ----------------------
+      
+      ERROR 1: directory "renv-test-ecd07ec4e426" contains 28 files; consider ignoring this directory
```
Comment (hadley) [inline tests/testthat/_snaps/dependencies.md:12]: I didn't look into this, but it's suspicious that the test writes 26 files and this reports 28.

Comment (hadley) [reply tests/testthat/_snaps/dependencies.md:12]: The extra files are `dependencies.R` and `renv`, which are not suspicious 😄

File: R/dependencies.R [hunk 2]
```diff
@@ -191,9 +191,8 @@ dependencies <- function(
 renv_dependencies_impl <- function(
   path = getwd(),
   root = NULL,
-  ...,
```
Comment (kevinushey) [inline R/dependencies.R:194]: Can we keep `...` to force following arguments to be named?

File: R/dependencies.R [hunk 3]
```diff
@@ -191,9 +191,8 @@ dependencies <- function(
...
   field = NULL,
-  progress = TRUE,
+  progress = FALSE,
```
Comment (kevinushey) [inline R/dependencies.R:195]: Doesn't this mean we no longer print progress reports for regular usages?

Comment (hadley) [reply R/dependencies.R:195]: No, because the default for `dependency()` is still `TRUE`

Comment (kevinushey) [reply R/dependencies.R:195]: Ohh got it!

File: R/dependencies.R (hunk 2 above)
Comment (kevinushey) [reply R/dependencies.R:194]: You may be over-estimating my discipline 😅

File: R/dependencies.R [hunk 4]
```diff
@@ -223,12 +222,6 @@ renv_dependencies_impl <- function(
   renv_dependencies_begin(root = root)
   defer(renv_dependencies_end())
 
-  dots <- list(...)
```
[the remark this answers is not in the document]
Comment (kevinushey) [reply R/dependencies.R:226]: I think we can drop this one; I don't think it's documented?

**Review by kevinushey — APPROVED** [review]

File: R/dependencies.R (hunk 2 above)
Comment (hadley) [reply R/dependencies.R:194]: Ok, I'll replace with `renv_dots_check`.

File: R/dependencies.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (kevinushey) [reply R/dependencies.R:227]: Oh oops. Should we just document it then?

**Pull request merged**